tests/test_macid.py

```diff
@@ -92,26 +103,44 @@ def test_get_all_pure_spe(self) -> None:
         self.assertTrue(np.array_equal(cpd_d2.values, np.array([[0, 1], [1, 0]])))
 
         macid = modified_taxi_competition()
-        all_spe = macid.get_all_pure_spe()
+        all_spe = macid.get_all_spe()
         self.assertTrue(len(all_spe) == 2)
 
         macid = prisoners_dilemma()
-        all_spe = macid.get_all_pure_spe()
+        all_spe = macid.get_all_spe()
         self.assertTrue(len(all_spe) == 1)
 
         macid = battle_of_the_sexes()
-        all_spe = macid.get_all_pure_spe()
+        all_spe = macid.get_all_spe()
         self.assertTrue(len(all_spe) == 2)
 
         macid3 = basic_different_dec_cardinality()
-        all_spe = macid3.get_all_pure_spe()
+        all_spe = macid3.get_all_spe()
         spe = all_spe[0]
         joint_policy = macid3.policy_profile_assignment(spe)
         cpd_d1 = joint_policy["D1"]
         cpd_d2 = joint_policy["D2"]
         self.assertTrue(np.array_equal(cpd_d1.values, np.array([0, 1])))
         self.assertTrue(np.array_equal(cpd_d2.values, np.array([[0, 0], [1, 0], [0, 1]])))
 
+        macid = battle_of_the_sexes()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 3)
+        macid = prisoners_dilemma()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 1)
+        macid = taxi_competition()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 1)
+        macid = modified_taxi_competition()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 2)
+        macid = basic_different_dec_cardinality()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 1)
+
+        # TODO add more tests for mixed equilibria
+
```

Comment:
In the appendix of our reasoning about causality in games paper here: https://causalincentives.com/  we give an example of a MACIM that has a NE in mixed policies (i.e. mixed strategy equiv), but not one in behavioural policies (our stochasticfunctionCPDs here). Because we don't represent them, could you implement this example and test that it throws an error
One can also add a check for sufficient recall in the method - if the MAIM has sufficient recall, then there must be at least one NE (and SPE) in behavioural policies